app/src/main/java/ch/epfl/balelecbud/BasicActivity.java

```diff
@@ -1,18 +1,97 @@
 package ch.epfl.balelecbud;
 
+import android.content.Intent;
+import android.view.MenuItem;
+
 import androidx.annotation.VisibleForTesting;
+import androidx.appcompat.app.ActionBarDrawerToggle;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.view.GravityCompat;
+import androidx.drawerlayout.widget.DrawerLayout;
+
+import com.google.android.material.navigation.NavigationView;
 
 import ch.epfl.balelecbud.authentication.Authenticator;
 import ch.epfl.balelecbud.authentication.FirebaseAuthenticator;
+import ch.epfl.balelecbud.friendship.SocialActivity;
 import ch.epfl.balelecbud.util.database.DatabaseWrapper;
 import ch.epfl.balelecbud.util.database.FirestoreDatabaseWrapper;
 
-public class BasicActivity extends AppCompatActivity {
+public class BasicActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {
```

Comment:
Should be abstract since we should not instance it

Reply:
Yes you're right